product_multiple_barcodes/models/product_barcode_multi.py

```diff
@@ -9,3 +9,7 @@ class ProductBarcodeMulti(models.Model):
 
     name = fields.Char('Barcode')
     product_id = fields.Many2one('product.product', string='Product')
+
+    _sql_constraints = [
```

Comment:
Instead of SQL constraint, you need to add python constraint. Because barcode should be unique in case you place it in fields:
Barcode
Additional Barcodes

Now you only verify that Additional Barcodes do not repeat. But on test server I was able to create the following combination:
Product 1: Barcode = test3, Additinal Barcodes - test
Product 2: Barcode = test

And it worked fine. But should not